packages/hash/api/src/graphql/resolvers/knowledge/entity/entity.ts

```diff
@@ -183,6 +189,198 @@ export const getAllLatestEntitiesResolver: ResolverFn<
   return entitySubgraph as Subgraph;
 };
 
+export const aggregateEntitiesResolver: Extract<
+  QueryResolvers<LoggedInGraphQLContext>["aggregateEntities"],
+  Function
+> = async (
+  _,
+  {
+    operation,
+    constrainsValuesOn,
+    constrainsPropertiesOn,
+    constrainsLinksOn,
+    constrainsLinkDestinationsOn,
+    isOfType,
+    hasLeftEntity,
+    hasRightEntity,
+  },
+  { dataSources },
+  __,
+) => {
+  const { graphApi } = dataSources;
+
+  /** @todo - should we throw an error, or should we just ignore the constraint and hope for the best */
+  if (
+    (operation.pageNumber !== undefined && operation.pageNumber !== null) ||
+    (operation.itemsPerPage !== undefined && operation.itemsPerPage !== null)
+  ) {
+    throw new ApolloError(
+      "Pagination on aggregateEntities results is not currently supported",
+    );
+  }
+
+  if (operation.multiSort !== undefined && operation.multiSort !== null) {
+    throw new ApolloError(
+      "Sorting on aggregateEntities results is not currently supported",
+    );
+  }
+
+  /** @todo - Encapsulate this transpilation in a shared function somewhere */
+  const filter: Filter = {
+    all: [
+      {
+        equal: [{ path: ["version"] }, { parameter: "latest" }],
+      },
+      {
+        equal: [{ path: ["archived"] }, { parameter: false }],
+      },
+    ],
+  };
+
+  if (operation.entityTypeId) {
+    filter.all.push({
+      equal: [
+        { path: ["type", "versionedUri"] },
+        { parameter: operation.entityTypeId },
```

Comment:
I believe @TimDiekmann said it will _just_ work with `STARTS_WITH`, yes. Alternatively we could add some hard-coded logic to actively spot a type-ID related query joined with a "STARTS_WITH" as structural queries actually support a "base URI" parameter